TableCollection.assert_equals reports the wrong reason

If two table collections differ in content *and* their top-level metadata differ, then `assert_equals(..., ignore_ts_metadata=True)` correctly raises an AssertionError, but incorrectly says it's because the metadata differ (which is true but not the reason). Consider:
```
t1 = tskit.TableCollection(sequence_length=1.0)
t2 = tskit.TableCollection(sequence_length=1.0)
t1.assert_equals(t2)
t1.metadata = b''
t2.metadata = b'abc'
t1.assert_equals(t2, ignore_ts_metadata=True)
t1.edges.add_row(0, 1, 0, 1)
t1.assert_equals(t2, ignore_ts_metadata=True)
```
gets no problems until the last line, when it gets
```
Traceback (most recent call last):
  File "<python-input-65>", line 8, in <module>
    t1.assert_equals(t2, ignore_ts_metadata=True)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/peter/projects/tskit-dev/tsvenv/lib/python3.13/site-packages/tskit/tables.py", line 3250, in assert_equals
    super().assert_equals(other)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^
  File "/home/peter/projects/tskit-dev/tsvenv/lib/python3.13/site-packages/tskit/metadata.py", line 1041, in assert_equals
    raise AssertionError(
        f"Metadata differs: self={self.metadata} " f"other={other.metadata}"
    )
AssertionError: Metadata differs: self=b'' other=b'abc'
```
This is not a big deal, but confusing.

--- a/python/CHANGELOG.rst
+++ b/python/CHANGELOG.rst
@@ -12,6 +12,12 @@
   associated with each individual as a numpy array.
   (:user:`benjeffery`, :pr:`3153`)
 
+
+**Fixes**
+
+- Correct assertion message when tables are compared with metadata ignored.
+  (:user:`benjeffery`, :pr:`3162`, :issue:`3161`)
+  
 --------------------
 [0.6.3] - 2025-04-28
 --------------------

--- a/python/tests/test_tables.py
+++ b/python/tests/test_tables.py
@@ -3758,6 +3758,25 @@ class TestTableCollection:
         with pytest.raises(TypeError):
             t1.equals(t2, True)
 
+        # When two tables differ, check that the right reason is given
+        t1 = tskit.TableCollection(sequence_length=1.0)
+        t2 = tskit.TableCollection(sequence_length=1.0)
+        t1.assert_equals(t2)
+        t1.metadata = b""
+        t2.metadata = b"abc"
+        t1.assert_equals(t2, ignore_ts_metadata=True)
+        t1.edges.add_row(0, 1, 0, 1)
+        with pytest.raises(AssertionError, match="EdgeTable number of rows"):
+            t1.assert_equals(t2, ignore_ts_metadata=True)
+        t2.metadata = b""
+        t2.edges.add_row(0, 1, 0, 1, metadata=b"abc")
+        t1.assert_equals(t2, ignore_metadata=True)
+        t1.edges.add_row(0, 1, 0, 1)
+        with pytest.raises(AssertionError, match="EdgeTable number of rows"):
+            t1.assert_equals(t2, ignore_metadata=True)
+        with pytest.raises(AssertionError, match="EdgeTable row 0 differs"):
+            t1.assert_equals(t2)
+
     def test_sequence_length(self):
         for sequence_length in [0, 1, 100.1234]:
             tables = tskit.TableCollection(sequence_length=sequence_length)

--- a/python/tskit/tables.py
+++ b/python/tskit/tables.py
@@ -3246,7 +3246,7 @@ class TableCollection(metadata.MetadataProvider):
         ):
             return
 
-        if not ignore_metadata or ignore_ts_metadata:
+        if not (ignore_metadata or ignore_ts_metadata):
             super().assert_equals(other)
 
         if not ignore_reference_sequence:

